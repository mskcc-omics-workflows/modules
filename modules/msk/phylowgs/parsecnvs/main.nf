process PHYLOWGS_PARSECNVS {
    tag "$meta.id"
    label 'process_low'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/phylowgs:v1.5-msk':
        'docker.io/mskcc/phylowgs:v1.5-msk' }"

    input:
    tuple val(meta), path(facetsgenelevel)

    output:
    tuple val(meta), path("cnvs.txt"), emit: cnv
    path "versions.yml"              , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    parse_cnvs.py \\
        ${args} \\
        ${facetsgenelevel}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        phylowgs: \$PHYLOWGS_TAG
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch cnvs.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        phylowgs: \$PHYLOWGS_TAG
    END_VERSIONS
    """
}
