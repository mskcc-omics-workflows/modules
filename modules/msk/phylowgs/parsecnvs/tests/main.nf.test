nextflow_process {

    name "Test Process PHYLOWGS_PARSECNVS"
    script "../main.nf"
    process "PHYLOWGS_PARSECNVS"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "phylowgs"
    tag "phylowgs/parsecnvs"

    test("phylowgs_parsecnvs - txt") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['facets_gene_level_txt'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }

            )
        }

    }

    test("phylowgs_parsecnvs - txt - stub") {

    options "-stub"

    when {
        process {
            """
            input[0] = [
                [ id:'test', single_end:false ], // meta map
                file('facets_gene_level_txt')
                ]
            """
        }
    }

    then {
        assertAll(
            { assert process.success },
            { assert snapshot(process.out).match() }

        )
    }

}


}
