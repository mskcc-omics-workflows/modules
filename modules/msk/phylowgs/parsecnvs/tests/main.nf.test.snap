{
    "phylowgs_parsecnvs - txt": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "cnvs.txt:md5,5ba8c1c06170c6d7ee9c18026b710154"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3fad79b7762271d747946e18f665089c"
                ],
                "cnv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "cnvs.txt:md5,5ba8c1c06170c6d7ee9c18026b710154"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3fad79b7762271d747946e18f665089c"
                ]
            }
        ],
        "timestamp": "2024-06-11T18:02:28.983981"
    },
    "phylowgs_parsecnvs - txt - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "cnvs.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3fad79b7762271d747946e18f665089c"
                ],
                "cnv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "cnvs.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3fad79b7762271d747946e18f665089c"
                ]
            }
        ],
        "timestamp": "2024-06-11T18:02:36.26591"
    }
}